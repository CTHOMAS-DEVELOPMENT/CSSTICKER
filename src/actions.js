export const addPlayer = player => ({
  type: "ADD PLAYER",
  payload: player
});
export const removePlayer = playerSet => ({
  type: "REMOVE PLAYER",
  payload: playerSet
});
export const updatePlayer = (player, index) => ({
  type: "UPDATE PLAYER",
  payload: player,
  index: index
});
export const updateDice = dice => ({
  type: "UPDATE DICE",
  payload: dice
});
