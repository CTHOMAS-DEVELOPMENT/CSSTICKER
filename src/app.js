import React from "react";
import { render } from "react-dom";
import { Provider } from "react-redux";
import "./styles/styles.scss";
import store from "./store";
import AppRouter from "./AppRouter";

render(
  <Provider store={store}>
    <AppRouter />
  </Provider>,
  document.getElementById("app")
);
