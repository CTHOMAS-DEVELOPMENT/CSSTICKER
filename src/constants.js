export const INITIALIZE = {
  dice: [0, 0],
  players: [
    {
      current: 0,
      total: 0,
      active: true
    },
    {
      current: 0,
      total: 0,
      active: false
    }
  ]
};
export const WINNINGSCORE = 100;
