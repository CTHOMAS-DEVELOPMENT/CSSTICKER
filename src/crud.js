import store from "./store";
import { BASETICKERURL, CURRENCYLIST } from "./constants";
import { addCryptoValues } from "./actions";
import axios from "axios";

export const cryptoCurrencyData = (bitcoin, bitcoinName) => 
{
  
  const url=`${ BASETICKERURL }fsym=${ bitcoin }&tsyms=${ CURRENCYLIST }`;
  axios.get( url )
      .then(res => {
        const cryptos= res.data;
        cryptos[bitcoin]=bitcoinName;
        store.dispatch(addCryptoValues(cryptos));
      })
      .catch(function (error) {
        // handle error
        console.log("ERROR : ",error);
      });
    }
/////
//ADD
/////
////////
//UPDATE
////////
////////
//DELETE
////////
