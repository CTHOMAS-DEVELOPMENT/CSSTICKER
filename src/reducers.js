import { INITIALIZE } from "./constants";
const initialState = INITIALIZE;
const rootReducer = (state = initialState, action) => {
  switch (action.type) {
    case "UPDATE DICE":
      return { ...state, dice: [...state.dice, action.payload] };
    case "ADD PLAYER":
      return { ...state, players: [...state.players, action.payload] };
    case "REMOVE PLAYER":
      return { ...state, players: action.payload };
    case "UPDATE PLAYER":
      return { ...state, players: [...state.players, action.payload] };
    default:
      return state;
  }
};
export default rootReducer;
