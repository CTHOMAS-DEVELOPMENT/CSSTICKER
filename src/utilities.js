//Utility functions that need no imports (Other than constants)

import { WINNINGSCORE } from "./constants";
export const getCurrentPlayer = players => {
  for (var i = 0; i < players.length; i++) {
    if (players[i].active) {
      return i;
    }
  }
  return 0;
};
export const getNextPlayer = (players, currentIndex) => {
  return currentIndex + 1 === players.length ? 0 : currentIndex + 1;
};
export const getRoleResult = (r1, r2, current, aggregate) => {
  //2 ones (Snake eyes) thrown
  if (r1 === 1 && r2 === 1) {
    return "SNAKE EYES";
  }
  //1 thrown
  if (r1 === 1 || r2 === 1) {
    return "LOSE SCORE";
  }
  //Winning score reached
  if (r1 + r2 + current + aggregate >= WINNINGSCORE) {
    return "WINNER";
  }
  return "AGGREGATE";
};
export const isStillInitialized = existingPlayers => {
  //Note: To be initialized, no player must hold any score and player 1 should be "active"
  let noScoreHeld = true;
  for (let i = 0; i < existingPlayers.length; i++) {
    if (existingPlayers[i].current !== 0 || existingPlayers[i].total !== 0) {
      noScoreHeld = false;
      break;
    }
  }

  return existingPlayers[0].active && noScoreHeld;
};
