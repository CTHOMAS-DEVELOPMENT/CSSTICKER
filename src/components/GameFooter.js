import React from "react";
import {
  faDice,
  faHandPaper,
  faSyncAlt
} from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { getCurrentPlayer } from "../utilities";
const GameFooter = props => {
  //Find current player
  let currentPlayerIndex = getCurrentPlayer(props.players);
  const player = props.players[currentPlayerIndex];
  //if current score of player is 0 set state to 0 for hold available
  const divStyle = {
    color: "yellow",
    fontSize: "3em"
  };
  return (
    <div>
      <div className="panelitem footerbottom">
        <div className="wrapper">
          <div className="box b">
            <button
              className="btn"
              title="Roll the dice"
              onClick={() => {
                props.role();
              }}
            >
              <FontAwesomeIcon icon={faSyncAlt} className="svgicon" />
              <FontAwesomeIcon icon={faDice} className="svgicon iconstatus" />
            </button>

            {player.current !== 0 && (
              <button
                className="btn"
                title="Hold"
                onClick={() => {
                  props.hold();
                }}
              >
                <FontAwesomeIcon icon={faHandPaper} className="svgicon" />
              </button>
            )}
          </div>
        </div>
        <div className="box d sbtxtsz_mini">
          {props.message ? props.message : ""}
        </div>
      </div>
    </div>
  );
};

export default GameFooter;
