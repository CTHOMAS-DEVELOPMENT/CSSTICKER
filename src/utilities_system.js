import { updatePlayer } from "./actions";
import { getNextPlayer } from "./utilities";

export const initializePlayerSet = players => {
  players[0].active = true;
  players[0].current = 0;
  players[0].total = 0;
  updatePlayer(players[0], 0);

  for (let i = 1; i < players.length; i++) {
    players[i].active = false;
    players[i].current = 0;
    players[i].total = 0;
    updatePlayer(players[i], 0);
  }
  return players;
};
export const setHolder = (players, index) => {
    let player = players[index];
    //Set the player to non active
    player.total += player.current;
    player.current = 0;
    player.active = false;
    //Set the next player to active
    index = getNextPlayer(players, index);
    player = players[index];
    player.active = true;
}
export const setPlayerTotalLossScore = (player, players, index) => {
  let message = `Player ${index +
    1} : You rolled two 1's and lose all your points.`;
  player.current = 0;
  player.total = 0;
  player.active = false;
  updatePlayer(player, index);
  
  //Set the active player
  index = getNextPlayer(
    players,
    index
  );
  player = players[index];
  player.active = true;
  updatePlayer(player, index);
  return message;
}
export const setPlayerLossScore = (player, players, index) => {
  let message = `Player ${index + 1} : You rolled a 1 and lose your score.`;
  player.current = 0;
  player.active = false;
  updatePlayer(player, index);

  //Set the active player
  index = getNextPlayer(players, index);
  player = players[index];
  player.active = true;
  updatePlayer(player, index);

  return message;
};
