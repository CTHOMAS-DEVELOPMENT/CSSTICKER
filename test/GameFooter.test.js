import React from "react";
import { shallow } from "enzyme";
import GameFooter from "../src/components/GameFooter";
import { INITIALIZE } from "../src/constants";

test("Should render GameFooter correctly..", () => {
  const wrapper = shallow(<GameFooter players={INITIALIZE.players} />);
  expect(wrapper).toMatchSnapshot();

});
