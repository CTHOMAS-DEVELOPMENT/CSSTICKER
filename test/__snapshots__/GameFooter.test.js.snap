// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render GameFooter correctly.. 1`] = `
<div>
  <div
    className="panelitem footerbottom"
  >
    <div
      className="wrapper"
    >
      <div
        className="box b"
      >
        <button
          className="btn"
          onClick={[Function]}
          title="Roll the dice"
        >
          <FontAwesomeIcon
            border={false}
            className="svgicon"
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  512,
                  512,
                  Array [],
                  "f2f1",
                  "M370.72 133.28C339.458 104.008 298.888 87.962 255.848 88c-77.458.068-144.328 53.178-162.791 126.85-1.344 5.363-6.122 9.15-11.651 9.15H24.103c-7.498 0-13.194-6.807-11.807-14.176C33.933 94.924 134.813 8 256 8c66.448 0 126.791 26.136 171.315 68.685L463.03 40.97C478.149 25.851 504 36.559 504 57.941V192c0 13.255-10.745 24-24 24H345.941c-21.382 0-32.09-25.851-16.971-40.971l41.75-41.749zM32 296h134.059c21.382 0 32.09 25.851 16.971 40.971l-41.75 41.75c31.262 29.273 71.835 45.319 114.876 45.28 77.418-.07 144.315-53.144 162.787-126.849 1.344-5.363 6.122-9.15 11.651-9.15h57.304c7.498 0 13.194 6.807 11.807 14.176C478.067 417.076 377.187 504 256 504c-66.448 0-126.791-26.136-171.315-68.685L48.97 471.03C33.851 486.149 8 475.441 8 454.059V320c0-13.255 10.745-24 24-24z",
                ],
                "iconName": "sync-alt",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
          <FontAwesomeIcon
            border={false}
            className="svgicon iconstatus"
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  640,
                  512,
                  Array [],
                  "f522",
                  "M592 192H473.26c12.69 29.59 7.12 65.2-17 89.32L320 417.58V464c0 26.51 21.49 48 48 48h224c26.51 0 48-21.49 48-48V240c0-26.51-21.49-48-48-48zM480 376c-13.25 0-24-10.75-24-24 0-13.26 10.75-24 24-24s24 10.74 24 24c0 13.25-10.75 24-24 24zm-46.37-186.7L258.7 14.37c-19.16-19.16-50.23-19.16-69.39 0L14.37 189.3c-19.16 19.16-19.16 50.23 0 69.39L189.3 433.63c19.16 19.16 50.23 19.16 69.39 0L433.63 258.7c19.16-19.17 19.16-50.24 0-69.4zM96 248c-13.25 0-24-10.75-24-24 0-13.26 10.75-24 24-24s24 10.74 24 24c0 13.25-10.75 24-24 24zm128 128c-13.25 0-24-10.75-24-24 0-13.26 10.75-24 24-24s24 10.74 24 24c0 13.25-10.75 24-24 24zm0-128c-13.25 0-24-10.75-24-24 0-13.26 10.75-24 24-24s24 10.74 24 24c0 13.25-10.75 24-24 24zm0-128c-13.25 0-24-10.75-24-24 0-13.26 10.75-24 24-24s24 10.74 24 24c0 13.25-10.75 24-24 24zm128 128c-13.25 0-24-10.75-24-24 0-13.26 10.75-24 24-24s24 10.74 24 24c0 13.25-10.75 24-24 24z",
                ],
                "iconName": "dice",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
        </button>
      </div>
    </div>
    <div
      className="box d sbtxtsz_mini"
    />
  </div>
</div>
`;
